source/base/data_out_base.cc

```diff
@@ -7718,19 +7718,15 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
       std::stringstream ss;
       DataOutBase::write_vtu_header(ss, vtk_flags);
       header_size = ss.str().size();
-      ierr = MPI_File_write(fh,
+      ierr = MPI_File_write_shared(fh,
                             DEAL_II_MPI_CONST_CAST(ss.str().c_str()),
                             header_size,
                             MPI_CHAR,
                             MPI_STATUS_IGNORE);
       AssertThrowMPI(ierr);
     }
 
-  ierr = MPI_Bcast(&header_size, 1, MPI_UNSIGNED, 0, comm);
-  AssertThrowMPI(ierr);
-
-  ierr = MPI_File_seek_shared(fh, header_size, MPI_SEEK_SET);
-  AssertThrowMPI(ierr);
+  
```

Comment:
I have to admit that I do not understand the calls used here. Can you add a comment that explains what `MPI_File_write_shared` does? In particular, is it a write on the root processor couples with a seek on all processes?

Reply:
Pengfei, can you add a comment in front of the write_shared that says
```
// Write the header on rank 0 and automatically move the 
// shared file pointer to the location after the header
```

Reply:
And yes, @bangerth , it is kind of interesting how the library will synchronize movement of the shared file pointer without the other ranks participating in a collective call. It works, though. ;-)

Reply:
That was the root of my confusion -- the other processes don't even call anything...

Reply:
Sure, @tjhei ,that's reasonable. 

Thanks for the explanation!